recipes/szip/all/conanfile.py

```diff
@@ -102,16 +102,27 @@ def _create_cmake_module_alias_targets(module_file, targets):
                     set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
                 endif()
             """.format(alias=alias, aliased=aliased))
-        tools.save(module_file, content)
+        content += """\
+                set (SZIP_FOUND TRUE)
+                set (SZIP_LIBRARIES szip::szip)
```

Comment:
this does not exist 😕

Reply:
This is my attempt to tackle this problem: https://github.com/conan-io/conan-center-index/pull/13551#issuecomment-1286358620
would appreciate some guidance ...

Reply:
> So, how do I write package_info() if I would like to support for both Conan v2 and v1:
> 
> * `find_package(szip CONFIG)` and `target_link_libraries(thingy szip::szip)`   (because it seems right)
> * `find_package(szip CONFIG)` and `target_link_libraries(thingy szip-shared)` (or szip-static, depending on what was built)
> * `find_package(SZIP)` and `target_link_libraries(thingy ${SZIP_LIBRARIES})`  (support legacy)
> * `find_package(SZIP)` and `target_link_libraries(thingy szip-shared)` (or szip-static, depending on what was built)

Out of all of these only the second one is "official"

As you've pointed out, we do not want to set the direction, szip is unmainteed AFAIK so there's no way to add to it at this point

Reply:
Ok, so just patch HDF5 (and others) to link to whichever target is available with the `if (TARGET szip-shared)` trick in the test_package